nodewatcher/nodewatcher.py

```diff
@@ -282,15 +284,17 @@ def _poll_instance_status(config, scheduler_module, asg_name, hostname, instance
                         continue
 
                     try:
-                        _self_terminate(asg_name, asg_conn, instance_id)
-                        termination_in_progress = True
+                        succeeded = _self_terminate(asg_name, asg_conn, instance_id)
+                        if succeeded:
+                            sys.exit(0)
+                        idletime = 0
```

Comment:
I think we shouldn't reset the idletime when the `_self_terminate` fails, because we have to retry to terminate the instance in the next round.

Reply:
if it fails with exception we do not reset the idletime. This gets reset only if we skip termination due to _maintain_size which means the node became essential to the cluster